packages/synapse-react-client/src/components/TimelinePlot/TimelinePlot.tsx

```diff
@@ -0,0 +1,243 @@
+import React, { useState } from 'react'
+import { useGetFullTableQueryResults } from '../../synapse-queries'
+import { BUNDLE_MASK_QUERY_RESULTS } from '../../utils/SynapseConstants'
+import hardcodedPhasesQueryResponseData, {
+  phaseObservationIndex,
+  phaseSpeciesIndex,
+} from './phasesQueryResponseData'
+import TimelinePhase from './TimelinePhase'
+import getColorPalette from '../ColorGradient/ColorGradient'
+import { Box } from '@mui/system'
+import {
+  ColumnSingleValueFilterOperator,
+  ColumnSingleValueQueryFilter,
+  Row,
+} from '@sage-bionetworks/synapse-types'
+import { ObservationCardSchema } from '../row_renderers/ObservationCard'
+import {
+  SQLOperator,
+  getAdditionalFilters,
+  parseEntityIdFromSqlStatement,
+} from '../../utils/functions'
+import { SizeMe } from 'react-sizeme'
+import TimelineLegendItem from './TimelineLegendItem'
+import { Skeleton } from '@mui/material'
+import TimelinePlotSpeciesSelector from './TimelinePlotSpeciesSelector'
+import NoContentAvailable from '../SynapseTable/NoContentAvailable'
+
+const OBSERVATION_PHASE_COLUMN_NAME = 'phase'
+const OBSERVATION_TIME_COLUMN_NAME = 'time'
+const OBSERVATION_TIME_UNITS_COLUMN_NAME = 'timeunits'
+const OBSERVATION_SUBMITTER_NAME_COLUMN_NAME = 'submittername'
+const OBSERVATION_TEXT_COLUMN_NAME = 'text'
+const OBSERVATION_TYPE_COLUMN_NAME = 'tag'
+const OBSERVATION_SUBMITTER_USER_ID_COLUMN_NAME = 'submitteruserid'
+
+export type TimelinePlotProps = {
+  sql: string
+  searchParams?: Record<string, string>
+  sqlOperator?: SQLOperator
+  defaultSpecies?: string //for test
+}
+export const TimelinePlot = ({
+  sql,
+  searchParams,
+  sqlOperator,
+  defaultSpecies,
+}: TimelinePlotProps) => {
+  // Fetch the table data
+  const eventsTableId = parseEntityIdFromSqlStatement(sql)
+  const [species, setSpecies] = useState<string | undefined | null>(
+    defaultSpecies,
+  )
+  const queryFilters =
+    getAdditionalFilters(eventsTableId, searchParams, sqlOperator) ?? []
+  const speciesFilter: ColumnSingleValueQueryFilter | undefined = species
+    ? {
+        columnName: 'species',
+        concreteType:
+          'org.sagebionetworks.repo.model.table.ColumnSingleValueQueryFilter',
+        values: [species],
+        operator: ColumnSingleValueFilterOperator.EQUAL,
+      }
+    : undefined
+  const additionalFilters = [...queryFilters]
+  if (speciesFilter) {
+    additionalFilters.push(speciesFilter)
+  }
+  const eventTableQuery = useGetFullTableQueryResults(
+    {
+      entityId: eventsTableId,
+      query: {
+        sql: `${sql} WHERE observationTime IS NOT NULL`,
+        sort: [
+          {
+            column: 'observationTime',
+            direction: 'ASC',
+          },
+        ],
+        additionalFilters: additionalFilters,
+      },
+
+      partMask: BUNDLE_MASK_QUERY_RESULTS,
+      concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
+    },
+    {
+      enabled: !!species,
```

Comment:
I don't know if this actually works.  It should be limited to the resource ID anyway, so this should not be a large result if it doesn't work.  Ideally it would only execute when it could apply the filters by both species and resourceId